modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java

```diff
@@ -111,8 +111,8 @@ public static TargetRequest create(MessageContext msgContext,
             }
 
 			String cType = getContentType(msgContext, configuration.isPreserveHttpHeader(HTTP.CONTENT_TYPE));
-			if (cType != null && (!httpMethod.equals("GET") && !httpMethod.equals("DELETE"))) {
-				String messageType = (String) msgContext.getProperty("messageType");
+			if (cType != null && (!httpMethod.equals("GET") && !RelayUtils.isDeleteRequestWithoutPayload(msgContext))) {
```

Comment:
Fixed in https://github.com/wso2/wso2-synapse/pull/801/commits/b6c89cf2bdf920a6e2a44ab6b3f1fbc2fabcf145